database/database.go

```diff
@@ -50,7 +50,7 @@ var drivers = make(map[string]Driver)
 
 // Driver is a function that opens a Datastore specified by its database driver type and specific
 // configuration.
-type Driver func(RegistrableComponentConfig) (Datastore, error)
+type Driver func(cfg RegistrableComponentConfig, passwordRequired bool) (Datastore, error)
```

Comment:
I added `passwordRequired bool` instead of making it a possible field in `RegistrableComponentConfig` as I do not want users to adjust this setting (this is part of Scanner's ConfigMap). I wanted to hardcode where password should be required vs not